### PR TITLE
test: add tracing to comms via --tracing-enabled

### PR DESCRIPTION
**Pull request by stringhandler**
Description
---
Add tracing to comms to debug timings via the `--tracing-enabled` flag

Motivation and Context
---
It's currently difficult to understand the timings of network calls and errors in the application.

How Has This Been Tested?
---
Tested manually

### CODE REVIEW

File: comms/src/protocol/messaging/protocol.rs [hunk 1]
```diff
@@ -247,6 +247,7 @@ impl MessagingProtocol {
         Ok(())
     }
 
+    // #[tracing::instrument(skip(self, out_msg), err)]
```
Comment (philipr-za) [inline comms/src/protocol/messaging/protocol.rs:250]: Maybe remove if its not being used but 🤷‍♂️

File: comms/src/protocol/rpc/client.rs [hunk 2]
```diff
@@ -494,14 +505,17 @@ where TSubstream: AsyncRead + AsyncWrite + Unpin + Send
 
         let (mut response_tx, response_rx) = mpsc::channel(10);
         if reply.send(response_rx).is_err() {
-            debug!(target: LOG_TARGET, "Client request was cancelled.");
+            event!(Level::WARN, "Client request was cancelled");
```
Comment (philipr-za) [inline comms/src/protocol/rpc/client.rs:508]: I believe these two lines are equivalent? The warn! is an event with this level so would remove one of these lines.

Comment (stringhandler) [reply comms/src/protocol/rpc/client.rs:508]: I didn't change all of the logs/warns to the tracing crate. I actually worried that we end up losing some logs, so I didn't take a blanket approach of swapping them out. At some point we should do it, but in the mean time there may be a period of duplicate logs

**Review by philipr-za — APPROVED** [review]
LGTM

**Pull request merged**